src/tools/rbd/action/Bench.cc

```diff
@@ -254,17 +254,18 @@ int do_bench(librbd::Image& image, io_type_t io_type,
     b.wait_for(io_threads - 1);
     i = 0;
     while (i < io_threads && off < io_bytes) {
+      uint64_t old_offs = thread_offset[i];
       if (random) {
         thread_offset[i] = (rand() % (size / io_size)) * io_size;
       } else {
         thread_offset[i] += io_size;
         if (thread_offset[i] + io_size > size)
           thread_offset[i] = 0;
       }
-
-      if (!b.start_io(io_threads, thread_offset[i], io_size, op_flags))
+      if (!b.start_io(io_threads, thread_offset[i], io_size, op_flags)) {
+        thread_offset[i] = old_offs;
         break;
-
+      }
```

Comment:
@ifed01 lgtm, Just a suggestion to consider though.
Don't you think the code would look a little nicer if instead of `old_offs` we introduced a variable that would store the new calculated offset (e.g. `thread_off`)? Then instead of updating `thread_offset[i]` before `start_io`, we would set this `thread_off`, pass it to `start_io`, and update `thread_offset[i]` onfly if `start_io` succeeded.